plugins/user-authenticators/ldap/src/org/apache/cloudstack/api/command/LinkDomainToLdapCmd.java

```diff
@@ -54,11 +54,11 @@
     @Parameter(name = ApiConstants.TYPE, type = CommandType.STRING, required = true, description = "type of the ldap name. GROUP or OU")
     private String type;
 
-    @Parameter(name = ApiConstants.LDAP_DOMAIN, type = CommandType.STRING, required = true, description = "name of the group or OU in LDAP")
+    @Parameter(name = ApiConstants.LDAP_DOMAIN, type = CommandType.STRING, required = false, description = "name of the group or OU in LDAP")
```

Comment:
Was this a regression that was added due to LDAP related enhancements, or could this cause issue if the field is made non-mandatory?

Reply:
the validation of the field is done deeper in the stack. A check is done to see if either ldapdomain or name is filled.